src/helper.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) DELTACAST.TV. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at * * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "helper.hpp"
+
+#include <thread>
+#include <optional>
+#include <VideoMasterCppApi/to_string.hpp>
+#include <VideoMasterCppApi/exception.hpp>
+#include <VideoMasterCppApi/helper/sdi.hpp>
+
+using namespace Deltacast::Wrapper;
+using namespace Deltacast::Wrapper::Helper;
+
+std::ostream& operator<<(std::ostream& os, Board& board)
+{
+    os << "\t" << "Board " << board.index() << ":  [ " << board.name() << " ]" << std::endl;
+    os << "\t" << "\t" << "- " << board.number_of_rx() << " RX / " << board.number_of_tx() << " TX" << std::endl;
+    os << "\t" << "\t" << "- Driver: " << board.driver_version() << std::endl;
+    os << "\t" << "\t" << "- PCIe ID: " << board.pcie_identifier() << std::endl;
+    os << "\t" << "\t" << "- SN: " << board.serial_number() << std::endl;
+    auto [ pcie_bus, number_of_lanes ] = board.pcie();
+    os << "\t" << "\t" << "- " << to_pretty_string(pcie_bus) << ", " << number_of_lanes << " lanes" << std::endl;
+
+    os << std::hex;
+    os << "\t" << "\t" << "- Firmware: 0x" << board.fpga().version() << std::endl;
+    if (board.has_scp())
+        os << "\t" << "\t" << "- SCP: 0x" << board.scp().version() << std::endl;
+    os << std::dec;
+
+    return os;
+}
+
+namespace Application::Helper
+{
+    std::optional<std::reference_wrapper<BoardComponents::Loopback>> get_loopback(Board& board, unsigned int channel_index)
+    {

```

Comment:
I don't see were we change the loopback state. Should we keep it?

Reply:
I forgot to add the enable/disable of loopback in the main. Just done it.